### PR TITLE
Feat/repay bundle

### PR DESCRIPTION
**Pull request by JoaquinBattilana**

### CODE REVIEW

**Comment by JoaquinBattilana** [comment]
Link T-5846

File: packages/contract-helpers/src/v3-pool-contract-bundle/index.ts [hunk 1]
```diff
@@ -329,5 +333,105 @@ export class PoolBundle
...
+        populatedTx.data = txData;
+        populatedTx.gasLimit = BigNumber.from(
+          gasLimitRecommendations[ProtocolAction.repayWithPermit].recommended,
+        );
+        return populatedTx;
+      },
+    };
+
+    this.repayWithATokensTxBuilder = {
+      generateTxData: ({ rateMode, user, amount, reserve }) => {
```
Comment (grothem) [inline packages/contract-helpers/src/v3-pool-contract-bundle/index.ts:412]: I think we need to also include `useOptimizedPath` here, and if it's true, use the `l2PoolService` to generate the tx data

File: packages/contract-helpers/src/lendingPool-contract-bundle/index.ts [hunk 2]
```diff
@@ -191,5 +197,41 @@ export class LendingPoolBundle
         return actionTx;
       },
     };
+    this.repayTxBuilder = {
```
Comment (grothem) [inline packages/contract-helpers/src/lendingPool-contract-bundle/index.ts:200]: Need to handle amount being `-1` here and convert it to max uint if it is

File: packages/contract-helpers/src/v3-pool-contract-bundle/index.ts [hunk 3]
```diff
@@ -329,5 +333,105 @@ export class PoolBundle
         return actionTx;
       },
     };
+
+    this.repayTxBuilder = {
```
Comment (grothem) [inline packages/contract-helpers/src/v3-pool-contract-bundle/index.ts:337]: Need to handle amount being `-1` here and convert it to max uint if it is

File: packages/contract-helpers/src/v3-pool-contract-bundle/index.ts (hunk 1 above)
Comment (grothem) [reply packages/contract-helpers/src/v3-pool-contract-bundle/index.ts:412]: also need to handle the `-1` case here

**Comment by grothem** [comment]
Link T-7615

**Review by grothem — APPROVED** [review]

**Pull request merged**